### PR TITLE
Upgrades Fantomas from v5 to v6

### PR DESCRIPTION
**Pull request by NickDarvey**
Upgrades Fantomas from v5 to v6. 

## Motivation
1. Getting the parameter name 'intonation' in the following example:
   ```fsharp
   type Cat =
    abstract meow: intonation:string -> string
   ```
   https://github.com/fsprojects/fantomas/pull/2510

## Changes

1. Namespace
   https://github.com/fsprojects/fantomas/blob/41fe8d62f212d9b2ab786c2a5873f0980e12975e/docs/docs/end-users/UpgradeGuide.md?plain=1#L83
1. Comma ranges in more places
   https://github.com/fsprojects/fantomas/pull/2880
1. Identifiers
   ?
1. Strict mode no longer used
    https://github.com/fsprojects/fantomas/blob/41fe8d62f212d9b2ab786c2a5873f0980e12975e/docs/docs/end-users/UpgradeGuide.md?plain=1#L78
1. ASTtransformer now depends on commas being present because of its use of List.zip in `mkTuple` and `mkSynLongIdent`.
1. ASTtransformer expects a range in a few places where it was optional before.

## TODO
- [x] Unknown issue omitting namespaces
- [x] Unknown issue making everything 'inline'
- [x] Record construction is empty
- [x] Rhs of with expression is empty

---

I suggest ignoring whitespace when viewing diff of the generated files.

https://github.blog/2018-05-01-ignore-white-space-in-code-review/

### CODE REVIEW

File: test/Myriad.IntegrationPluginTests/ArbitaryFile.fs
Comment (NickDarvey) [inline test/Myriad.IntegrationPluginTests/ArbitaryFile.fs:1]: Note the omitted namespaces and everything 'inline' in this file.

File: test/Myriad.IntegrationPluginTests/FieldsLensDus.fs [hunk 1]
```diff
@@ -1,257 +1,250 @@
...
+
+namespace rec TestFields
+
+module Test1 =
+    open Input
+    let one (x: Test1) = x.one
+    let two (x: Test1) = x.two
+    let three (x: Test1) = x.three
+    let four (x: Test1) = x.four
+    let create (one: int) (two: string) (three: float) (four: float32) : Test1 = { }
```
Comment (NickDarvey) [inline test/Myriad.IntegrationPluginTests/FieldsLensDus.fs:185]: Note record construction is empty

File: test/Myriad.IntegrationPluginTests/FieldsLensDus.fs [hunk 2]
```diff
@@ -1,257 +1,250 @@
...
+
+    let map
+        (mapone: int -> int)
+        (maptwo: string -> string)
+        (mapthree: float -> float)
+        (mapfour: float32 -> float32)
+        (record': Test1)
+        : Test1 =
+        { record' with }
```
Comment (NickDarvey) [inline test/Myriad.IntegrationPluginTests/FieldsLensDus.fs:194]: Note rhs of with expression is empty

**Comment by 7sharp9** [comment]
Thanks!

**Comment by 7sharp9** [comment]
Thanks, I'll try to release a new nuget soon.

**Pull request merged**